src/core/service.c

```diff
@@ -2958,9 +3018,19 @@ static int service_stop(Unit *u) {
 
 static int service_reload(Unit *u) {
         Service *s = ASSERT_PTR(SERVICE(u));
+        int r;
 
         assert(IN_SET(s->state, SERVICE_RUNNING, SERVICE_EXITED));
 
+        /* If we have confexts extensions, try to reload vpick'd confext
+         * extensions, which is particularly beneficial for notify-reload
+         * services that could potentially pick up a new version of its
+         * configuration.
+         */
+        r = service_reload_extensions(u);
+        if (r < 0)
+                log_unit_error_errno(u, r, "Failed to reload confexts, ignoring: %m");
```

Comment:
Please move this into `service_enter_reload`, which does all the heavy lifting and real state changes. And this should be at LOG_WARNING.